lib/cloud_controller/dea/app_stager_task.rb

```diff
@@ -80,8 +103,8 @@ def handle_first_response(response, error, promise)
 
       def handle_second_response(response, error)
         @multi_message_bus_request.ignore_subsequent_responses
-        ensure_staging_is_current!
-        check_staging_error!(response, error)
+        ensure_staging_is_current!(false)
```

Comment:
checking the task_id is a nit and costs us nothing other than more code to make it not happen.
Again, not sure why I care if it is nats or http. we should always verify the task_id. Don't assume the path above you.

One can start an application, have staging start, stop the app, start the app, and the previous stage may not complete so you still need to verify that you are the current task_id even with NATs.